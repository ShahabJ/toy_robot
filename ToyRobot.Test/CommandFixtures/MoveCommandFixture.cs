using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using ToyRobot.Library.Commands;
using ToyRobot.Library.Model;
using ToyRobot.Library.Service;
using Xunit;

namespace ToyRobot.Test
{
    public class MoveCommandFixture
    {
        private CommandBase command;
        private GenericRobot robot;
        private GenericTable table;
        public MoveCommandFixture()
        {
            command = new MoveCommand();
            table = new GameTable();
            robot = new GameRobot(table);
        }

        [Theory]
        [InlineData("MOVE")]
        [InlineData("move")]
        [InlineData("Move")]
        public void IsValidCommandShouldReturnTrueFor(string commandName)
        {
            command.IsValidCommand(commandName).Should().BeTrue();
        }

        [Theory]
        [InlineData("Move ")]
        [InlineData(" move")]
        [InlineData("move 1")]
        [InlineData("LEFT")]
        [InlineData("EVOM")]
        public void IsValidCommandShouldReturnFalseFor(string commandName)
        {
            command.IsValidCommand(commandName).Should().BeFalse();
        }

        [UseReporter(typeof(DiffReporter))]
        [Fact]
        public void ExecuteReport()
        {   
            var result = new List<string>();
            var positions = new List<Position>();
            
            positions.Add(new Position(0, 0, DirectionEnum.NORTH));
            positions.Add(new Position(0, 0, DirectionEnum.EAST));
            positions.Add(new Position(1, 0, DirectionEnum.WEST));
            positions.Add(new Position(0, 1, DirectionEnum.SOUTH));

            foreach (var position in positions)
            {
                robot.CurrentPosition = position;
                string currentPosition = robot.CurrentPosition.ToString();
                string newPosition = command.Execute(robot).CurrentPosition.ToString();
                result.Add($"Position before move:{currentPosition}, Position After move:{newPosition}");
            }

            Approvals.VerifyAll(result,"Verifyinng Move Commnad => ");
        }
    }
}
