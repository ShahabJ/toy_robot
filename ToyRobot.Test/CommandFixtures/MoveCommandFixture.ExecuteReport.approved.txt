Verifyinng Move Commnad => [0] = Position before move:0,0,NORTH, Position After move:0,1,NORTH
Verifyinng Move Commnad => [1] = Position before move:0,0,EAST, Position After move:1,0,EAST
Verifyinng Move Commnad => [2] = Position before move:1,0,WEST, Position After move:0,0,WEST
Verifyinng Move Commnad => [3] = Position before move:0,1,SOUTH, Position After move:0,0,SOUTH
